app/renderer/actions/Session.js

```diff
@@ -751,3 +766,18 @@ function addCustomCaps (caps) {
     ...(platformName.toLowerCase() === 'ios' ? iosCustomCaps : {}),
   };
 }
+
+export function bindWindowClose () {
+  return (dispatch, getState) => {
+    window.addEventListener('beforeunload', async (evt) => {
+      const {driver} = getState().inspector;
+      if (driver) {
+        try {
+          const action = quitSession('Window closed');
+          await action(dispatch, getState);
+        } catch (ign) {}
```

Comment:
should we also set driver to null?

Reply:
it doesn't matter; when the window is closed all these JS objects will be destroyed